src/ui/viewmodels/MemoryViewerViewModel.hh

```diff
@@ -235,6 +239,8 @@ private:
     static std::unique_ptr<ra::ui::drawing::ISurface> s_pFontSurface;
     static std::unique_ptr<ra::ui::drawing::ISurface> s_pFontASCIISurface;
     static int s_nFont;
+    static std::list<ra::ByteAddress> s_listMemViewHistory;
+    static std::list<ra::ByteAddress>::iterator s_iterMemViewHistoryIndex;
```

Comment:
`std::vector` is generally preferred over `std::list` if you're not inserting stuff in the middle. `std::list` uses node pointers to make updates faster at the cost of memory. `ra::ByteAddress` is a 32-bit value, so the overhead of the node is quite expensive.

Also, there's no reason for these to be static. You wouldn't want your history shared across multiple instances - And I'm not really sure how this will interact with the pointer finder. You might be okay since you're managing the addition of items to the history from the inspector.